packages/liveblocks-node/src/thread-notification-helpers.ts

```diff
@@ -0,0 +1,284 @@
+    mentionedUserId: userId,
+  });
+
+  if (lastUnreadCommentWithMention !== null) {
+    return {
+      type: "unreadMention",
+      comments: [lastUnreadCommentWithMention],
+    };
+  }
+
+  return {
+    type: "unreadReplies",
+    comments: unreadComments,
+  };
+}
+
+export type ResolveRoomInfoArgs = {
+  /**
+   * The ID of the room to resolve
+   */
+  roomId: string;
+};
+
+export type GetThreadNotificationUnreadCommentsDataOptions<
+  U extends BaseUserMeta = DU,
+> = {
+  /**
+   * Which format to transform the comment body to.
+   */
+  format?: "html" | "json";
+  /**
+   * A function that returns info from user IDs.
+   */
+  resolveUsers?: (
+    args: ResolveUsersArgs
+  ) => OptionalPromise<(U["info"] | undefined)[] | undefined>;
+  /**
+   * A function that returns room info from room IDs.
+   */
+  resolveRoomInfo?: (
+    args: ResolveRoomInfoArgs
+  ) => OptionalPromise<DRI | undefined>;
+};
+
+export type UnreadCommentAuthorData = {
+  id: string;
+  name: string;
+  avatar?: string;
+};
+
+/** @internal */
+export const getAuthor = async <U extends BaseUserMeta = DU>({
+  userId,
+  resolveUsers,
+}: {
+  userId: string;
+  resolveUsers?: (
+    args: ResolveUsersArgs
+  ) => OptionalPromise<(U["info"] | undefined)[] | undefined>;
+}): Promise<UnreadCommentAuthorData> => {
+  const fallback: UnreadCommentAuthorData = {
+    id: userId,
+    name: userId,
+  };
+  if (resolveUsers) {
+    const users = await resolveUsers({ userIds: [userId] });
+    return users?.[0] && users?.[0].name
+      ? { id: userId, name: users[0].name }
+      : fallback;
+  }
+  return fallback;
+};
+
+export type UnreadCommentData = {
+  id: string;
+  threadId: string;
+  roomId: string;
+  author: UnreadCommentAuthorData;
+  createdAt: Date;
+  body: string | CommentBodyJson;
+  commentUrl?: string;
+  roomName: string;
+};
+
+export type UnreadRepliesData = {
+  type: "unreadReplies";
+  roomName: string;
+  comments: UnreadCommentData[];
+};
+export type UnreadMentionData = {
+  type: "unreadMention";
+  comments: UnreadCommentData[];
+  roomName: string;
+};
+export type UnreadCommentsData = (UnreadRepliesData | UnreadMentionData) & {
+  roomName: string;
+};
+
+/**
+ *
+ * Get unread comments from a `ThreadNotificationEvent`
+ * It returns either an object containing a list of unread replies for a thread
+ * or either an object containing a list of the last unread comment where the notification
+ * receiver was mentioned in.
+ *
+ * @param params.client Liveblocks node client
+ * @param params.event The thread notification event
+ * @returns An unread comments object
+ *
+ * @example Unread replies:
+ * {
+ *  type: "unreadReplies",
+ *  comments: [unread_comment_0, unread_comment_1, ...],
+ *  roomName: "acme"
+ * }
+ *
+ * @example Unread mention:
+ * {
+ *  type: "unreadMention",
+ *  comments: [unread_comment_with_mention],
+ *  roomName: "acme"
+ * }
+ */
+export async function getThreadNotificationUnreadComments(params: {
```

Comment:
I think this function should be named similarly to `getThreadNotificationData` given that it returns the same type of things but with a higher level

Reply:
I do not understand here 🤔 
What name do you see for this function then?

Reply:
Something like `getThreadNotificationFormattedData` or similar, to me that's the only difference between the two functions (the shape of the data it returns), `getThreadNotificationData` also returns "unread comments"

Reply:
Regarding your comment here and I think it's the good place to use the word `Resolved` no? 
 I would see something like
 ```ts
export async function getThreadNotificationResolvedData(): Promise<ResolvedCommentsData>
 ```
 
 WDYT of this new approach here?